### PR TITLE
Add hostoperator configmap

### PR DESCRIPTION
**Pull request by tinakurian**
adding configMap for host operator as discussed here: codeready-toolchain/host-operator#238 (comment)

this adds the registration service url

related to: https://github.com/codeready-toolchain/toolchain-infra/pull/26

### CODE REVIEW

File: pkg/configuration/configuration.go [hunk 1]
```diff
@@ -124,11 +131,62 @@ func loadFromSecret(cl client.Client) error {
 	return nil
 }
 
+// loadFromConfigMap retrieves the host operator configMap
+func loadFromConfigMap(cl client.Client) error {
+	// get the configMap name
+	configMapName := getResourceName("HOST_OPERATOR_CONFIG_NAME")
```
Comment (alexeykazakov) [inline pkg/configuration/configuration.go:137]: Should be `HOST_OPERATOR_CONFIG_MAP_NAME` , right?
```suggestion
	configMapName := getResourceName("HOST_OPERATOR_CONFIG_MAP_NAME")
```

File: pkg/configuration/configuration_test.go [hunk 2]
```diff
@@ -125,6 +125,65 @@ func TestLoadFromSecret(t *testing.T) {
...
+	t.Run("default", func(t *testing.T) {
+		// when
+		config := getDefaultConfiguration(t)
+
+		// then
+		assert.Equal(t, "https://registration.crt-placeholder.com", config.GetRegistrationServiceURL())
+	})
+	t.Run("env overwrite", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_CONFIG_NAME", "test-config")
```
Comment (alexeykazakov) [inline pkg/configuration/configuration_test.go:140]:
```suggestion
		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_CONFIG_MAP_NAME", "test-config")
```

File: pkg/configuration/configuration_test.go [hunk 3]
```diff
@@ -125,6 +125,65 @@ func TestLoadFromSecret(t *testing.T) {
...
+		// test env vars are parsed and created correctly
+		regServiceURL := os.Getenv("HOST_OPERATOR_REGISTRATION_SERVICE_URL")
+		assert.Equal(t, regServiceURL, "test-url")
+		testTest := os.Getenv("HOST_OPERATOR_TEST_TEST")
+		assert.Equal(t, testTest, "test-test")
+	})
+
+	t.Run("configMap not found", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_CONFIG_NAME", "test-config")
```
Comment (alexeykazakov) [inline pkg/configuration/configuration_test.go:172]:
```suggestion
		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_CONFIG_MAP_NAME", "test-config")
```

File: pkg/configuration/configuration.go (hunk 1 above)
Comment (tinakurian) [reply pkg/configuration/configuration.go]: good catch! thanks @alexeykazakov fixed here: eab40417d9d02d007255ae0c75da627258b2a03b

File: pkg/configuration/configuration_test.go (hunk 2 above)
Comment (tinakurian) [reply pkg/configuration/configuration_test.go]: fixed here: eab4041

File: pkg/configuration/configuration_test.go (hunk 3 above)
Comment (tinakurian) [reply pkg/configuration/configuration_test.go]: fixed here: eab4041

**Pull request merged**